lib/functions/main/check.sh

```diff
@@ -1,62 +1,77 @@
 #!/usr/bin/env bash
 
+###############################################################################
+#
+# string_is_contain()
+# Check for the presence of a word in a string if the field separator is mixed.
+# IFS can be "," comma " " space or ", " both or "\n" the end of a line.
+# usage: if string_is_contain "word1 word2,word3" "word2"; then
+#
+string_is_contain() {
+	[[ -z "$@" ]] || [[ "$#" != 2 ]] && \
+	echo "err: ${FUNCNAME[0]}: Bud argument $@" >&2 && return 1
+	local list=${1// /\\n}
+	echo -e ${list//,/\\n} | grep -q "^$2$"
+}
+
 ###############################################################################
 #
 # build_task_is_enabled()
 #
-# $1: _taskNameToCheck - a single task name to check for BUILD_ONLY enablement
+# $1 - a single task name to check for BUILD_ONLY enablement
 # return:
-#   0 - if BUILD_ONLY is empty or if the task name is listed by BUILD_ONLY
+#   0 - if BUILD_ONLY if the task name is listed by BUILD_ONLY
 #   1 - otherwise 
 #
 build_task_is_enabled() {
-	# remove all "
-	local _taskNameToCheck=${1//\"/}
-	local _buildOnly=${BUILD_ONLY//\"/}
-	# An empty _buildOnly allows any taskname
-	[[ -z $_buildOnly ]] && return 0
-	_buildOnly=${_buildOnly//,/ }
-	for _buildOnlyTaskName in ${_buildOnly}; do
-		[[ "$_taskNameToCheck" == "$_buildOnlyTaskName" ]] && return 0
-	done
-	return 1
+	string_is_contain "$BUILD_ONLY" "$1"
 }
 
 ###############################################################################
 #
 # build_validate_buildOnly()
 #
 # This function validates the list of task names defined by global
-# variable BUIDL_ONLY versus the locally defined constant
+# variable BUILD_ONLY versus the locally defined constant
 # list __all_valid_buildOnly.
 #
 # In case of future extensions, please maintain the list of valid task names
 # only here.
 #
 build_validate_buildOnly() {
+	[[ -z $BUILD_ONLY ]] && display_alert "BUILD_ONLY has empty value" "call dialog" "err" && exit 1
+	local _buildOnly="${BUILD_ONLY}"
+
 	# constant list of all valid BUILD_ONLY task names - can be :comma: or :space: separated
-	local _all_valid_buildOnly="u-boot kernel armbian-config armbian-zsh plymouth-theme-armbian armbian-firmware armbian-bsp chroot bootstrap"
-	# remove all "
-	local _buildOnly=${BUILD_ONLY//\"/}
+	local _build_packages="u-boot,kernel,armbian-config,armbian-zsh,plymouth-theme-armbian,armbian-firmware,armbian-bsp"
```

Comment:
Refer to my comment above regarding single place of constant definitions:
Assign `_build_packages=$(CONST_build_packages)`